extensions/pkg/controller/worker/templates/machineclasses.tpl.yaml

```diff
@@ -54,13 +54,44 @@ spec:
                   must be unique.
                 type: string
             type: object
+            x-kubernetes-preserve-unknown-fields: true
           kind:
             description: 'Kind is a string value representing the REST resource this
                   object represents. Servers may infer this from the endpoint the client
                   submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
             type: string
           metadata:
             type: object
+          nodeTemplate:
+            description: NodeTemplate contains subfields to track all node resources
+              and other node info required to scale nodegroup from zero
+            properties:
+              capacity:
+                additionalProperties:
+                  anyOf:
+                  - type: integer
+                  - type: string
+                  pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                  x-kubernetes-int-or-string: true
+                description: Capacity contains subfields to track all node resources
+                  required to scale nodegroup from zero
+                type: object
+              instanceType:
+                description: Instance type of the node belonging to nodeGroup
+                type: string
+              region:
+                description: Region of the expected node belonging to nodeGroup
+                type: string
+              zone:
+                description: Zone of the expected node belonging to nodeGroup
+                type: string
+            required:
+            - capacity
+            - instanceType
+            - region
+            - zone
+            type: object
+            x-kubernetes-preserve-unknown-fields: true
```

Comment:
Why is this line needed?

Reply:
Just to confirm: Is this because `capacity` is of type `corev1.ResourceList` which itself is just a `map` and can contain arbitrary keys?

Reply:
yes you are correct.

Reply:
OK, thanks!